### PR TITLE
ggml : fix odd blocks for ARM_NEON

### PR DESCRIPTION
**Pull request by ggerganov**
target #8549 

- Same fix as #8549 for `Q4_0`, `Q4_1`, `Q5_0`, `Q5_1` and `Q8_0`
- Fix `IQ4_NL` Metal kernel handling odd `ne01`
- Remove `Q4_0` SSE3 and Longarch special handling of first 2 blocks

---

- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [ ] Low
  - [ ] Medium
  - [ ] High

### CODE REVIEW

File: ggml/src/ggml-quants.c [hunk 1]
```diff
@@ -4076,18 +4020,18 @@ void ggml_vec_dot_q4_0_q8_0(int n, float * restrict s, size_t bs, const void * r
         acc_3 = _mm_add_ps(p3_d, acc_3);
     }
 
-    *s = hsum_float_4x4(acc_0, acc_1, acc_2, acc_3);
+    sumf = hsum_float_4x4(acc_0, acc_1, acc_2, acc_3);
 #elif defined(__riscv_v_intrinsic)
     float sumf = 0.0;
```
Comment (slaren) [inline ggml/src/ggml-quants.c:4025]: `sumf` may be redefined for riscv, here and in the other functions.

**Comment by ggerganov** [comment]
Thanks, will merge after https://github.com/ggerganov/llama.cpp/pull/8549 is merged

**Comment by slaren** [comment]
@ggerganov the target of this PR is #8549, so I believe this would need to be merged first. Do you intend to create a new PR targeting master afterwards?

**Comment by ggerganov** [comment]
Correct, it's better to merge in #8549 first - will update now

**Review by slaren — APPROVED** [review]

**Pull request merged**